### PR TITLE
IgnoreCase = false for BackupStorageRedundancy

### PR DESCRIPTION
**Pull request by xaliciayang**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by xaliciayang** [comment]
BackupStorageRedundancy (introduced in S175 (2020-09-22) Ignite) was supposed to NOT ignore case; this check was mistakenly removed from **[this PR](https://github.com/azure/azure-powershell/pull/12934)**.  This is important because [AzureSqlDatabaseAdapter.cs](https://github.com/Azure/azure-powershell/blob/bd7456be622b65b6a2257b795fa706c6d9d116d5/src/Sql/Sql/Database/Services/AzureSqlDatabaseAdapter.cs#L399) and [AzureSqlDatabaseModel.cs](https://github.com/Azure/azure-powershell/blob/bd7456be622b65b6a2257b795fa706c6d9d116d5/src/Sql/Sql/Database/Model/AzureSqlDatabaseModel.cs#L301) are both case sensitive.

File: src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs [hunk 1]
```diff
@@ -205,7 +205,7 @@ public class NewAzureSqlDatabase : AzureSqlDatabaseCmdletBase<AzureSqlDatabaseCr
         /// </summary>
         [Parameter(Mandatory = false,
             HelpMessage = "The Backup storage redundancy used to store backups for the SQL Database. Options are: Local, Zone and Geo.")]
-        [ValidateSet("Local", "Zone", "Geo")]
+        [ValidateSet("Local", "Zone", "Geo", IgnoreCase = false)]
```
Comment (VeryEarly) [inline src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs:208]: this looks like a breaking change for me, do you want to release this in October major release?

**Review by VeryEarly — CHANGES_REQUESTED** [review]
please also add change log

File: src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs (hunk 1 above)
Comment (xaliciayang) [reply src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs:208]: I'd like this to go as soon as possible; this is added to a parameter, BackupStorageRedundancy, that was only introduced in the last milestone  (introduced in S175 (2020-09-22) Ignite).  @dingmeng-xue and @msJinLei have reviewed my previous changes that this belongs to. 

I will add change log too.

**Review by VeryEarly — APPROVED** [review]

File: src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs (hunk 1 above)
Comment (VeryEarly) [reply src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs:208]: merged by mistake, the change log should under upcoming release, I'll make another PR to fix it

**Pull request merged**